lua/autorun/server/sv_increase_starting_ammo.lua

```diff
@@ -1,22 +1,17 @@
+local blacklistedAmmo = {
+    ["Smoke Grenades"] = true,
+    ["Flash Grenades"] = true,
+    ["Frag Grenades"] = true,
+    ["40MM"] = true
+    -- we don't actually have to add RPG ammo here
+}
 
-local function givePlayerAmmoForWeapon( player, weapon )
-    local ammoType = weapon:GetPrimaryAmmoType()
-    local currentAmmo = player:GetAmmoCount( ammoType )
-    if ( currentAmmo == 0 ) then
-        player:GiveAmmo( 90, ammoType, false )
-    end
-end
-
-local function giveCW2AmmoOnSpawn( player )
-    timer.Simple( 0, function()
-        for _, weapon in pairs( player:GetWeapons() ) do
-            local isCW2Weapon = string.sub( weapon:GetClass(), 1, 2 ) == "cw"
-            if ( isCW2Weapon ) then
-                givePlayerAmmoForWeapon( player, weapon )
-            end
+function givePlayerCWAmmo( ply )
+    for ammoName, bulletInfo in pairs( CustomizableWeaponry.ammoTypes ) do
+        if not blacklistedAmmo[ammoName] then
+            ply:SetAmmo( 1000, ammoName )
```

Comment:
Might be easier and more efficient to generate the list of "whitelisted" ammo types on init.

Underneath the `blacklistedAmmo`, a simple:
```lua
local cwAmmo = {}
```

then a new hook,
```lua
hook.Add( "Initialize", "CFC_CW2_GetAmmoTypes", function()
    for ammoName in pairs( CustomizableWeaponry.ammoTypes ) do
        cwAmmo[ammoName] = not blacklistedAmmo[ammoName]
    end
end )
```

Then this section can be:

```suggestion
function givePlayerCWAmmo( ply )
    for ammoName in pairs( cwAmmo ) do
        ply:SetAmmo( 1000, ammoName )
    end
end
```